bundles/org.openhab.core.io.rest.sitemap/src/main/java/org/eclipse/smarthome/io/rest/sitemap/internal/PageChangeListener.java

```diff
@@ -231,18 +231,20 @@ public void changeStateTo(Item item, State state) {
                 event.valuecolor = itemUIRegistry.getValueColor(w);
                 event.widgetId = itemUIRegistry.getWidgetId(w);
                 event.visibility = itemUIRegistry.getVisiblity(w);
-                // event.item contains data from the item including its state (in event.item.state)
-                String widgetTypeName = w.eClass().getInstanceTypeName()
-                        .substring(w.eClass().getInstanceTypeName().lastIndexOf(".") + 1);
-                boolean drillDown = "mapview".equalsIgnoreCase(widgetTypeName);
-                Predicate<Item> itemFilter = (i -> i.getType().equals(CoreItemFactory.LOCATION));
-                event.item = EnrichedItemDTOMapper.map(item, drillDown, itemFilter, null, null);
+                if (!skipWidget) {
```

Comment:
Maybe instead of this we should do a ternary?
```
Item itemToBeSent = skipWidget ? item : w.getItem();
event.item = EnrichedItemDTOMapper.map(itemToBeSent, drillDown, itemFilter, null, null);
```

...
```